### ISSUE TITLE
set DeviceError at the correct times

### ISSUE THREAD

**Issue by killenb** [ChimeraTK/ApplicationCore#136]
Child of #123
Implement after #129

* The initial value of DeviceError must be set to 1 before the application goes into multi-threaded mode (see spec. 2.1)
* The value must not be set to 0 before the the initialisation is done and all recovery accessors have been written (see spec 2.3.5). This is currently happening too early.

* [x] Put the initial value at the correct place
* [x] Move setting the value to 0 to the correct place.

FIXME: How to test this race-condition free?

**Comment by killenb** [comment]
Fixes spec 11.1 and 11.3

**Comment by mhier** [comment]
This ticket is already done as part of #129 .
